### PR TITLE
Add GET /hosted-payments/{id} endpoint

### PR DESCRIPTION
**Pull request by punit-shah-cko**
### Description of changes in PR

[PI-1355]:
- Add `id` & `_links.self` properties to `POST /hosted-payments` response
- Add `_links.self` property to `POST /payment-links` response

[PI-1358]:
- Add spec for new `GET /hosted-payments/{id}` endpoint

Also:
- Minor fixes for Payment Links spec

## Technical Writer

[//]: # 'Tag your Technical Writer'
@chrisi-webster-cko 

[PI-1355]: https://checkout.atlassian.net/browse/PI-1355?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PI-1358]: https://checkout.atlassian.net/browse/PI-1358?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by tom-finlayson-cko — DISMISSED** [review]

File: nas_spec/swagger.yaml [hunk 1]
```diff
@@ -46,12 +46,13 @@ info:
...
 
-     | Date          |      Description of change                                        |
-     |---------------|-------------------------------------------------------------------|
-     | 2021/11/11    |  Added `3ds.challenge_indicator` to card payment requests.        |
-     | 2021/11/03    |  Adds `identification` object under parent `sender` object in payment request. |
-     | 2021/10/18    |  Added the `marketplaces.sub-entities` object to support split payments.  |
-     | 2021/10/12    |  Added Hosted Payments Page and Payment Links to specification.   |
+      | Date       | Description of change                                                         |
+      | ---------- | ----------------------------------------------------------------------------- |
+      | 2021/11/?? | Added specification for "Get Hosted Payments Page details" endpoint.          |
```
Comment (ben-ahmady-cko) [inline nas_spec/swagger.yaml:51]: Please change the `??` when this is ready to go live 😅

File: abc_spec/swagger.yaml [hunk 2]
```diff
@@ -49,11 +49,12 @@ info:
...
     # Changelog
 
-     | Date          |      Description of change                                        |
-     |---------------|-------------------------------------------------------------------|
-     | 2021/11/11    |  Added `3ds.challenge_indicator` to card payment requests.        |
-     | 2021/10/18    |  Added `processing.purpose` to card payouts.                      |
-     | 2021/10/18    |  Added `recommendation_code` to payment response.                 |
+      | Date       | Description of change                                                |
+      | ---------- | -------------------------------------------------------------------- |
+      | 2021/11/?? | Added specification for "Get Hosted Payments Page details" endpoint. |
```
Comment (ben-ahmady-cko) [inline abc_spec/swagger.yaml:54]: Please change the `??` when this is ready to go live 😅

**Review by ben-ahmady-cko — APPROVED** [review]
Hey @punit-shah-cko the build looks good but I'm not familiar with the subject matter so relying on the approval you've had from Tom for that—and please make sure to change the changelog entries before merging 🙏

**Comment by chrisi-webster-cko** [comment]
@punit-shah-cko Does your minor fix relate to PI-1370?

**Comment by punit-shah-cko** [comment]
> @punit-shah-cko Does your minor fix relate to PI-1370?

@chrisi-webster-cko Not exactly - they're just minor changes to the existing spec (see the changes in commit d5e26dd):
- Made descriptions in the yaml file more readable (this doesn't change the output)
- Changed the `payment_id` field in the Get Payment Link details Active/Expired response examples to be `null` instead of the string `"undefined"`
  before: Screenshot 2021-11-15 at 14 49 25, after: Screenshot 2021-11-15 at 14 51 20
- Removed unnecessary parentheses from the regex pattern for the `PaymentLinkId` schema
- Show the `payment` and `payment_actions` links for the Get Payment Link details Payment Received response example only (and not the Active/Expired examples)
- Use the `PaymentLinkId` schema for the `id` property in the Create a Payment Link response
- Removed the `Cko-Request-Id` and `Cko-Version` headers from the Get Payment Link details response, as we don't actually include these headers in our response

**Review by ben-ahmady-cko — APPROVED** [review]
If Chrisi/Tom are happy, all good (Chrisi can't approve any more as a contributor)

**Pull request merged**